lib/atscppapi/src/Transaction.cc

```diff
@@ -433,14 +411,12 @@ class initializeHandles
   initializeHandles(GetterFunction getter) : getter_(getter) {}
   bool operator()(TSHttpTxn txn, TSMBuffer &hdr_buf, TSMLoc &hdr_loc, const char *handles_name)
   {
-    if (!hdr_buf && !hdr_loc) {
-      if (getter_(txn, &hdr_buf, &hdr_loc) == TS_SUCCESS) {
-        return true;
-      } else {
-        LOG_ERROR("Could not get %s", handles_name);
-      }
+    hdr_buf = NULL;
+    hdr_loc = NULL;
+    if (getter_(txn, &hdr_buf, &hdr_loc) == TS_SUCCESS) {
+      return true;
     } else {
-      LOG_ERROR("%s already initialized", handles_name);
+      LOG_ERROR("Could not get %s", handles_name);
```

Comment:
The handles inside the Request and Response objects need to be reset here as well, otherwise, plugins may reference invalid handles via those objects.